c++/src/kj/async-inl.h

```diff
@@ -2286,7 +2312,10 @@ class CoroutineBase::AwaiterBase {
 
 private:
   UnwindDetector unwindDetector;
-  OwnPromiseNode node;
+
+  // only one of these is set
```

Comment:
You might be able to get away with using `OwnPromiseNode` for both. The trick is that `OwnPromiseNode` disposes the `PromiseNode` by calling its `destroy()` method. In this particular case, you can have the implementation of `destroy()` do nothing.